### ISSUE TITLE
#[test_case("*o?a*r", "foobar")] fails

### ISSUE THREAD

**Issue by Lweb** [becheran/wildmatch#8]
**Describe the bug**
`*o?a*r` should match `foobar`. 
However, the result of `WildMatch::new("*o?a*r").matches("foobar")` is `false`.

**To Reproduce**
Add the following test case:
```
#[test_case("*o?a*r", "foobar")]
```
Run `cargo test`.

**Expected behavior**
Tests should pass.

**Desktop (please complete the following information):**
 - OS: Windows 10
 - Version: 21H2

**Additional context**
rustc version
```
rustc 1.61.0 (fe5b13d68 2022-05-18)
```

**Comment by becheran** [comment]
@Lweb Thanks for reporting. It is a bug. I will try to find a solution

**Comment by becheran** [comment]
Fixed with release `2.1.1`
